File: LeetCode_2038/Main.c
#include <stdbool.h>

bool winnerOfGame(char* colors) {
    int alice = 0, bob = 0;
    int cnta = 0, cntb = 0;

    for (int i = 0; colors[i] != '\0'; i++) {
        if (colors[i] == 'A') {
            cnta++;
            bob += cntb > 2 ? cntb - 2 : 0;
            cntb = 0;
        } else {
            cntb++;
            alice += cnta > 2 ? cnta - 2 : 0;
            cnta = 0;
        }
    }

    bob += cntb > 2 ? cntb - 2 : 0;
    alice += cnta > 2 ? cnta - 2 : 0;

    return alice > bob;
}

File: LeetCode_2038/Main.js
/**
 * @param {string} colors
 * @return {boolean}
 */
var winnerOfGame = function(colors) {
    let alice = 0, bob = 0;
    let cnta = 0, cntb = 0;

    for (let i = 0; i < colors.length; i++) {
        if (colors[i] === 'A') {
            cnta++;
            bob += Math.max(0, cntb - 2);
            cntb = 0;
        } else {
            cntb++;
            alice += Math.max(0, cnta - 2);
            cnta = 0;
        }
    }

    bob += Math.max(0, cntb - 2);
    alice += Math.max(0, cnta - 2);

    return alice > bob;
};
